dev-guide/test_framework.md

```diff
@@ -0,0 +1,219 @@
+
+### Testing Framework Scripts
+All of the testing framework scripts and associated data have been placed into their own Git repository at https://stash.pnnl.gov/scm/jgcri/gcam-testing-framework.git
+
+However it has been included in the gcam-core Git repository as a "Submodule" stored in "util/testing-framework".  A Git Submodule is essentially a version controlled pointer.  It allows us to say version AAA of GCAM requires version BBB of gcam-testing-framework.
+
+Thus, let's take as an example, if I create a branch in which I update the version of the BaseX library.  I can create a branch in the testing framework that will install the updated BaseX for testing as well; and update the submodule pointer to point to that branch in the testing framework.  In this way the tests from this pull request will run just fine with the updated version of BaseX.  And existing pull requests would not be affected so would continue to run just fine with the old version of BaseX.
+
+Most GCAM users will never need to actually "initialize" this submodule and so won't affect them.  I have configured Jenkins to "initialize" ALL submodules (which it would have had to do to get hector anyways) so it will get the testing scripts with it's checkout of the gcam-core.
+
+#### Jenkinsfile Script
+Jenkins allows us to define tests as arbitrary scripts in what it calls "Jenkinsfile" scripts.  This allows us lots of flexibility to do what we need.  The downside is the language these scripts are written in is Groovy, a Java based scripting language, which is yet another obscure language to learn.  Because all of the build steps are contained in these scripts configuring two test, triggered by Bitbucket and triggered monthly, is easy; they just need to call this script.
+
+#### The Configured Test
+
+#### Launch Windows Tests
+We have configured a "Pull Request Notifier" which is triggered on:
+* Pull request open
+* Pull request has commits added (`RESCOPED_FROM`)
+
+It will notify Jenkins with the "JGCRI-gcam-windows" tag with the meta data:
+* `PULL_REQUEST_FROM_BRANCH`
+* `PULL_REQUEST_ID`
+* `PULL_REQUEST_USER_DISPLAY_NAME`
+
+Jenkins will the trigger the [JGCRI/gcam-windows](https://ci.pnnl.gov/jenkins/job/JGCRI/job/gcam-windows/) test.  It will checkout the `$PULL_REQUEST_FROM_BRANCH` and run the script `util/testing-framework/windows/Jenkinsfile`. It will run on the `windows-build-02` build executor which already has the development tools we need in installed:
+* R 3.5
+* MsBulid 15.0 / Visual Studio 2017
+* Java 1.8
+
+From there we will need to perform the following steps:
+1. Build the `gcamdata` package to generate XMLs
+  * Create some space inside of our workspace to install the R packages
+  * Install the `devtools` package
+  * Install the `testthat` package for some reason (it is not required to just run gcamdata with no tests yet R refused to load gcamdata without it)
```

Comment:
Delete "for some reason" and maybe update the parentheses to say "this is required to load gcamdata, but tests are not required to be run" or something.